src/localplayer.cpp

```diff
@@ -627,12 +635,19 @@ v3s16 LocalPlayer::getStandingNodePos()
 }
 
 // Horizontal acceleration (X and Z), Y direction is ignored
-void LocalPlayer::accelerateHorizontal(const v3f &target_speed, const f32 max_increase)
+void LocalPlayer::accelerateHorizontal(const v3f &target_speed, const f32 max_increase, bool slippery)
 {
         if (max_increase == 0)
                 return;
 
         v3f d_wanted = target_speed - m_speed;
+	if (slippery)
+	{
+		if (target_speed == v3f(0))
+			d_wanted = -m_speed*.05;
+		else
+			d_wanted = target_speed*.1 - m_speed*.1;
```

Comment:
Code style sould be corrected a bit: `d_wanted = (target_speed - m_speed) * 0.1f;`